src/util.h

```diff
@@ -40,9 +40,9 @@ NORETURN void Fatal(const char* msg, ...);
 #elif defined(__cplusplus) && __cplusplus >= 201103L && defined(__clang__)
 #define NINJA_FALLTHROUGH [[clang::fallthrough]]
 #elif defined(__cplusplus) && __cplusplus >= 201103L && defined(__GNUC__) && \
-    __GNUC__ >= 7
+    __GNUC__ >= 7 && !defined(__INTEL_COMPILER)
```

Comment:
No version of the Intel compiler supports this attribute, but they claim to be a gcc? That seems... suboptimal, and personally I'd rather not see Ninja cluttered even more with special rules for every compiler.

Is there a version of the Intel compiler that would support the C++17 version of the fallthrough attribute?

Is there a specific macro that can be checked for support for the attribute, without consideration of compiler vender?

Reply:
Since Intel 13.0, Intel defines __GNUC__ to signify ABI compatibility.
What I didn't try is seeing if simply "fallthrough" can be used in general with current compilers.
Yes, now I specify that C++17 standard should be used, since not all compilers default to C++17. However, MSVC C++17 is too strict for Ninja at the moment, so I let MSVC stay at default.

Reply:
Intel [defines](https://software.intel.com/en-us/cpp-compiler-developer-guide-and-reference-additional-predefined-macros) `__cplusplus=1` instead of a specific C++ revision level

Reply:
Intel does not like the first option `#define NINJA_FALLTHROUGH [[fallthrough]]` so it's not useful to have

```cxx
#if defined(__cplusplus) && __cplusplus >= 201703L || defined(__INTEL_COMPILER)
```

Reply:
This is the value Intel icpc is happy with:

```cxx
#elif defined(__GNUC__) && __GNUC__ >= 7 // gcc 7 or Intel compiler
#define NINJA_FALLTHROUGH __attribute__ ((fallthrough))
```

Reply:
That's a bit odd since Intel [docs](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2016/p0188r1.pdf) say that `[[fallthrough]]` is supported...

Reply:
I fixed the macro `[[fallthrough]]` logic to be cleaner. From the docs noted just above, I think this is one of the cleanest ways to do this.